gulpfile.js

```diff
@@ -1,185 +1,113 @@
+var Promise = require('es6-promise').Promise;
+var source = require('vinyl-source-stream');
+var sourcemaps = require('gulp-sourcemaps');
+var uglify = require('gulp-uglify');
 
 gulp.task('lint', function() {
-   return gulp.src([
+   var sources = [
       path.join(__dirname, '/*.js'),
       path.join(__dirname, '/src/*.js'),
       path.join(__dirname, '/test/*.js')
-   ],
-   {
+   ];
+   var opts = {
       base: './'
-   })
-   .pipe(jshint())
-   .pipe(jscs({
-      fix: true
-   }))
-   .pipe(stylish.combineWithHintResults())
-   .pipe(jscs.reporter())
-   .pipe(jshint.reporter('jshint-stylish'))
-   .pipe(gulp.dest('.'));
-});
+   };
 
-gulp.task('test', function(done) {
-   runTests(true, false, done);
+   return gulp.src(sources, opts)
+      .pipe(jshint())
+      .pipe(jscs())
+      .pipe(stylish.combineWithHintResults())
+      .pipe(jscs.reporter())
+      .pipe(jshint.reporter('jshint-stylish'))
+      .pipe(gulp.dest('.'))
+      ;
 });
 
-gulp.task('test:ci', function(done) {
-   runTests(true, true, done);
+gulp.task('coverage', function() {
+   return gulp.src('src/**/*.js')
+      .pipe(istanbul())
+      .pipe(istanbul.hookRequire())
+      ;
 });
 
-gulp.task('test:auto', function(done) {
-   runTests(false, false, done);
+gulp.task('test:mocha', ['coverage'], function() {
+   var srcOpts = {
+      read: false
+   };
+   var mochaOpts = {
+      timeout: 30000,
+      slow: 5000
+   };
+
+   return gulp.src('test/test.*.js', srcOpts)
+      .pipe(mocha(mochaOpts))
+      .pipe(istanbul.writeReports())
+      ;
 });
-gulp.task('clean', function () {
-   return del('dist/*');
+
+gulp.task('clean', function() {
+   return Promise.all([del('dist/*'), del('coverage/*')]);
 });
 
 gulp.task('build', function() {
-   var browserifyInstance = browserify({
+   var bundler = browserify({
       debug: true,
       entries: 'src/github.js',
       standalone: 'Github'
    });
 
-   browserifyInstance
+   bundler
+      .transform('babelify')
       .bundle()
       .pipe(source('github.js'))
       .pipe(buffer())
       .pipe(sourcemaps.init({
          loadMaps: true
       }))
-            .pipe(uglify())
+      .pipe(uglify())
       .pipe(rename({
          extname: '.bundle.min.js'
       }))
       .pipe(sourcemaps.write('.'))
-      .pipe(gulp.dest('dist'));
+      .pipe(gulp.dest('dist'))
+      ;
+
+   gulp.src('src/github.js')
+      .pipe(babel())
+      .pipe(sourcemaps.init())
+      .pipe(sourcemaps.write('.'))
+      .pipe(gulp.dest('dist'))
+      ;
 
    return gulp.src('src/github.js')
+      .pipe(babel())
       .pipe(sourcemaps.init())
       .pipe(rename({
          extname: '.min.js'
       }))
       .pipe(uglify())
       .pipe(sourcemaps.write('.'))
-      .pipe(gulp.dest('dist'));
+      .pipe(gulp.dest('dist'))
+      ;
 });
 
 gulp.task('default', ['clean'], function() {
-   gulp.start('lint', 'test', 'build');
+   gulp.start('lint', 'test:mocha', 'build');
 });
-
-var sauceLaunchers = {
```

Comment:
It does remove sauce completely because for PRs Sauce doesn't work. The intent is to add it back for `master` so we can populate the browser matrix.